### ISSUE TITLE
Bootstrapping more-than-one Root Components of the same class with unique selectors

### ISSUE THREAD

**Issue by dgaeta** [angular/angular#11255]
**I'm submitting a ...**  (check one with "x")

```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Calling bootstrap on the same root Component Class more than once, each with a unique _selector_, results in each _ComponentRef._hostElement.nativeElement_ referencing only the first bootstrapped MetaData selector. The ComponentRef._hostElement.nativeElement does not change if the same Component class is bootstrapped with a different unique_selector_.

**Expected/desired behavior**
When a Component class is bootstrapped with a unique _selector_ in the MetaData (Component Decorator parameter), then each Component will reference the corresponding unique native element.

**Reproduction of the problem**
If the current behavior is a bug or you can illustrate your feature request better with an example, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).

```
export class AppComponent {
  constructor(private vcRef: ViewContainerRef, private resolver: ComponentResolver) {
    console.log('annotations');
    console.log(Reflect.getMetadata('annotations', AppComponent));

  }
}
```

```
bootstrap( Component({
  selector: 'my-app-' + id,
  encapsulation: ViewEncapsulation.Emulated,
  template: `<p>Dynamic Component {{message}}</p>`
 })(AppComponent)).then(app => {
    //app['_hostElement'].nativeElement =  `<div class="Ng2"><h1>Angular 2-${this.context.instanceId}</h2><my-app-${this.context.instanceId}/></div>`
    console.log(app['_hostElement'].nativeElement);
    console.log('Bootstrap Successful');
    console.log(app);
}, err => {
    console.error(err);
});

```

**What is the motivation / use case for changing the behavior?**
Using Angular2 components inside of a React Component. When a React component that use's Angular2's templating (and other nice functionalities), the first React Component bootstraps as desired. But when a second instance of the React component is created - bootstrap() is called on the same AppComponent class - but the nativeElement points to the first nativeElement that was initialized.

**Please tell us about your environment:**
- **Angular version:** 2.0.0-beta.17
- **Browser:** [all ] 
- **Language:** [ TypeScript X.X |  ES5]

**Comment by zoechi** [comment]
dup of #7136 ?

**Comment by DzmitryShylovich** [comment]
@dgaeta is this still an issue? could u pls update your repro?

**Comment by trotyl** [comment]
Closed by #15668
